src/ast_utils.h

```diff
@@ -46,6 +46,34 @@ struct BreakSeeker : public PostWalker<BreakSeeker, Visitor<BreakSeeker>> {
   }
 };
 
+// Finds all functions that are reachable via direct calls.
+
+struct DirectCallGraphAnalyzer : public PostWalker<DirectCallGraphAnalyzer, Visitor<DirectCallGraphAnalyzer>> {
+  Module *module;
+  std::vector<Function*> queue;
+  std::unordered_set<Function*> reachable;
+
+  DirectCallGraphAnalyzer(Module* module, std::vector<Function*> root) : module(module) {
```

Comment:
That's why some people write `Struct const *var;` instead of `const Struct *var;`, so in general you can just say that const modifies the thing to the left. Of course you can make it more complicated than that... :-)